script/cli/doc/export.lua

```diff
@@ -0,0 +1,354 @@
+local ws       = require 'workspace'
+local vm       = require 'vm'
+local guide    = require 'parser.guide'
+
+local getDesc  = require 'core.hover.description'
+local getLabel = require 'core.hover.label'
+local jsonb    = require 'json-beautify'
+local util     = require 'utility'
+local markdown = require 'provider.markdown'
+
+---@alias doctype
+---| 'doc.alias'
+---| 'doc.class'
+---| 'doc.field'
+---| 'doc.field.name'
+---| 'doc.type.arg.name'
+---| 'doc.type.function'
+---| 'doc.type.table'
+---| 'funcargs'
+---| 'function'
+---| 'function.return'
+---| 'global.type'
+---| 'global.variable'
+---| 'local'
+---| 'luals.config'
+---| 'self'
+---| 'setfield'
+---| 'setglobal'
+---| 'setindex'
+---| 'setmethod'
+---| 'tableindex'
+---| 'type'
+
+---@class docUnion broadest possible collection of exported docs, these are never all together.
+---@field [1] string in name when table, always the same as view
+---@field args docUnion[] list of argument docs passed to function
+---@field async boolean has @async tag
+---@field defines docUnion[] list of places where this is doc is defined and how its defined there
+---@field deprecated boolean has @deprecated tag
+---@field desc string code commentary
+---@field extends string | docUnion ? what type this 'is'. string:<Parent_Class> for type: 'type', docUnion for type: 'function', string<primative> for other type 's
+---@field fields docUnion[] class's fields

```

Comment:
some strange line ending here at **line 41-42** as well
github's diff is showing a `^M` at the end 😕 
![image](https://github.com/user-attachments/assets/57f65cb5-be2a-4dc0-8e94-51fba8810baa)

---

And with the whole codebase is using `LF` (pure `\n`) as line endings, I suggest to use `LF` for newly created files for consistency.